### ISSUE TITLE
Generated code not properly formatted, dart format complains

### ISSUE THREAD

**Issue by martin-formigas** [dart-lang/mockito#620]
We have 2 generated mocks that force us to manually format the file after each generation step to be able to pass CI `dart format` checks.

A `BeamerDelegate` from the `beamer` package:

```dart
@GenerateMocks(
  <Type>[
    BeamerDelegate,
  ],
)
class GeneratedMocks {}
```

gives us 

```dart
  @override
  set onPopPage(
          bool Function(
            _i3.BuildContext,
            _i3.Route<dynamic>,
            dynamic,
          )? _onPopPage) =>
      super.noSuchMethod(
        Invocation.setter(
          #onPopPage,
          _onPopPage,
        ),
        returnValueForMissingStub: null,
      );
```

which formatted looks like this:

```dart
  @override
  set onPopPage(
          bool Function(
    _i3.BuildContext,
    _i3.Route<dynamic>,
    dynamic,
  )?
              _onPopPage) =>
      super.noSuchMethod(
        Invocation.setter(
          #onPopPage,
          _onPopPage,
        ),
        returnValueForMissingStub: null,
      );
```

And a `Query` from the `graphql_codegen` package.

```dart
@GenerateMocks(
  <Type>[
    Query$Media$media$media,
  ],
)
class GeneratedMocks {}
```

which gives us this:

```dart
            returnValue: <_i67
                .Query$Media$media$media$files>[],
```

which formatted looks like this:

```dart
            returnValue: <
                _i67.Query$Media$media$media$files>[],
```

**Comment by srawlins** [comment]
Very curious. The code is formatted as it's last step, so there must be a skew between your `dart format` and your `dart_style` dependency.

Can you show a few things:

* `dart --version`
* the resolved version of `dart_style`, found in `.dart_tool/package_config.json`

**Comment by martin-formigas** [comment]
Sure thing. Here we go:

`Dart SDK version: 2.19.6 (stable) (Tue Mar 28 13:41:04 2023 +0000) on "windows_x64"`

```json
    {
      "name": "dart_style",
      "rootUri": "file:///C:/Users/goddc/AppData/Local/Pub/Cache/hosted/pub.dev/dart_style-2.3.0",
      "packageUri": "lib/",
      "languageVersion": "2.19"
    },
```

**Comment by srawlins** [comment]
It's possible the formatter improved the formatting in `dart_style-2.3.0`. I see in [the changelog](https://pub.dev/packages/dart_style/changelog):

> * Better indentation of multiline function types inside type argument lists.
> * Don't split after < in collection literals.

I think each of these items could explain the changes you posted above (thanks for the exact unexpected changes). So your `dart format` is using an older version of dart_style, reverting the formatting to a worse (presumably) style.

This will right itself when you bump to Dart 3.0.0 SDK. But if you want to fix it now, before Dart 3.0 is released, you can probably just add `dart_style: 2.2.5` to your `dev_dependencies`, to lock it to the last version before 2.3.0.

**Comment by martin-formigas** [comment]
You are absolutely right, that works perfectly. Thanks for your help and investigation!
